### PR TITLE
DeleteDashboard: Redirect to home after deleting a dashboard

### LINKED ISSUE #78810: Dashboards: Deleting a dashboard from settings is not redirecting to home
### What happened?

When deleting a dashboard, I receive a confirmation alert but it doesn’t take me to another place.

### What did you expect to happen?

Redirect to home or the dashboard list.

### Did this work before?

Yes, it's a regression

### How do we reproduce it?

1. Create a dashboard
2. Go to settings
3. Delete it

https://github.com/grafana/grafana/assets/20256983/45d9e0d8-d63f-4636-b2b7-636c89e08960

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: main
OS: OSX
Browser: Chrome

### Grafana platform?

I use Grafana Cloud

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by ivanortegaalba**
Fixes #78810

**Diagnosis**
The `useDeleteDashboard` was listening to when the result was present in the `state.value` from `useAsyncFn` giving a cb to be called `onDeleteDashboard`. The callback returned by `useAsyncFn` is used inside another `useAsyncFn` that hides the modal once the `onDeleteDashboard` is resolved.

Because `onDeleteDashboard` is resolved before the next re-render is triggered, the useEffect declared in `useDeleteDashboard` is never called, so the variables and the replace is never invoked.

**Solution** 
I just simplified this piece of code, since the custom hook `useDeleteDashboard` is never used. I approached moving everything into a single async callback called when the user clicks confirm. I also removed the duplicated notification to show only one since it did not add much value.

### CODE REVIEW

**Review by evictorero — COMMENTED** [review]
I've tested it and works as expected. Left some comments.

File: public/app/features/manage-dashboards/state/actions.ts [hunk 1]
```diff
@@ -360,7 +360,12 @@ export function getFolderById(id: number): Promise<{ id: number; title: string }
 }
 
 export function deleteDashboard(uid: string, showSuccessAlert: boolean) {
-  return getBackendSrv().delete<DeleteDashboardResponse>(`/api/dashboards/uid/${uid}`, { showSuccessAlert });
+  return getBackendSrv()
```
[the remark this answers is not in the document]
Comment (ivanortegaalba) [reply public/app/features/manage-dashboards/state/actions.ts:363]: Ups 🙈

**Comment by ivanortegaalba** [comment]
> I've tested it and works as expected. Left some comments.

Fixed, sorry

**Review by evictorero — APPROVED** [review]
LGTM! Shouldn't be backported since it is a regression?

**Review by dprokop — APPROVED** [review]
Works as expected, nice!

Did you manage to figure out when/what introduced this regression? Also, I think this may deserve a backport to 10.2.x

**Pull request merged**